src/R/Interpreters/Impl/Implementation/RInstallation.cs

```diff
@@ -82,13 +62,22 @@ public sealed class RInstallation {
         private IEnumerable<IRInterpreterInfo> GetCompatibleEnginesFromRegistry(ISupportedRVersionRange svr) {
             svr = svr ?? new SupportedRVersionRange();
             var engines = GetInstalledEnginesFromRegistry().Where(e => svr.IsCompatibleVersion(e.Version));
-            // Remove duplicates (MRC registers under multiple keys)
-            var mrc = engines.FirstOrDefault(e => e.Name.Contains("Microsoft"));
-            if(mrc != null) {
-                var dupes = engines.Where(e => e.InstallPath.EqualsIgnoreCase(mrc.InstallPath)).Except(new IRInterpreterInfo[] { mrc });
-                engines = engines.Except(dupes);
+            // Among duplicates by path (MRC registers under multiple keys) take the highest version
+            var microsoftEngines = engines.Where(e => e.Name.Contains("Microsoft") || e.InstallPath.Contains("Microsoft"));
+            var installLocations = microsoftEngines.Select(e => e.InstallPath.TrimTrailingSlash()).Distinct();
```

Comment:
Yes, something like that. I unrolled for easier debugging. If CRAN duplicates paths, then its fine. There problem here is that MRC/MRS update in place without deleting old keys. Not sure if we want to extend heuristics (in fact, fixing their bug) to everyone.